backend/src/routes/classroom.ts

```diff
@@ -90,4 +90,109 @@ export const classroomRoute = new Elysia({ prefix: '/classroom' })
                 thumbnail: t.File(),
             }),
         },
+    )
+    .post('/join-classroom', async ({ body, user, session, set }) => {
+        if (!user || !session) {
+            set.status = 401;
+            return {
+                status: 'error',
+                message: 'Unauthenticated, Please sign in and try again',
+            };
+        }
+
+        const joiningCode = (body as { classroomCode: string }).classroomCode;
+
+        const codeRecord = await sql`
+            SELECT classroom_id, expires_at, section
+            FROM classroom_invite_code
+            WHERE code = ${joiningCode}
+        `;
+
+        if (codeRecord.length == 0) {
+            return {
+                status: 'error',
+                message: 'There is no classroom, please recheck the code.',
+            };
+        }
+
+        const expiresAt = new Date(codeRecord[0].expires_at).getTime();
+        const currentTime = new Date().getTime();
+
+        if (expiresAt < currentTime) {
+            return {
+                status: 'error',
+                message: 'This code is expired, please contact your teacher.',
+            };
+        }
+
+        const studentSection = [codeRecord[0].section];
+
+        const userId = user.id;
+
+        const classroomId = codeRecord[0].classroom_id;
+
+        const isAlreadyJoined = await sql`
+            SELECT classroom_id, user_id
+            FROM study
+            WHERE classroom_id = ${classroomId} AND user_id = ${userId}
+        `;
+
+        if (isAlreadyJoined.length != 0) {
+            return {
+                status: 'error',
+                message: 'You have already joined the class.',
+            };
+        }
```

Comment:
So user can join 1 section / 1 classroom? Why we have to have section as array?

Reply:
removed section